docs/proposals/node-allocatable.md

```diff
@@ -0,0 +1,190 @@
+   unscheduled resources). This value is the focus of this proposal. See [below](#api-changes) for
+   more details.
+4. **System Reserved** (proposed) - Compute resources reserved for system components such as the docker daemon, kubelet, kube proxy, etc.
+
+### API changes
+
+#### Allocatable
+
+Add `Allocatable` (4) to
+[`NodeStatus`](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/HEAD/docs/api-reference/v1/definitions.html#_v1_nodestatus):
+
+```
+type NodeStatus struct {
+  ...
+  // Allocatable represents schedulable resources of a node.
+  Allocatable ResourceList `json:"allocatable,omitempty"`
+  ...
+}
+```
+
+Allocatable will be computed by the Kubelet (unless explicitly set, see [below](#allocatable-flag))
+and reported periodically to the API server. It is defined to be:
+
+```
+   [Allocatable] = [Node Capacity] - ( [Reserved] + [System Reserved] + [Kernel usage (smoothed)] )
+```
+
+The scheduler will use `Allocatable` in place of `Capacity` when scheduling pods, and the Kubelet
+will use it when performing admission checks.
+
+- Since kernel usage can fluctuate and is out of kubernetes control, `Allocatable` may vary over
+  time.
+
+#### System Reserved
+
+`SystemReserved` is the parameter specifying resources reserved for system components (4). It is
+provided as a command-line flag to the Kubelet at startup, and therefore cannot be changed during
+normal Kubelet operation (this will change in the [future](#future-work)).
+
+The flag will be specified as a serialized `ResourceList`, with resources defined by the API
+`ResourecName` and values specified in `resource.Quantity` format, e.g.:
+
+```
+--system-reserved=cpu=500m,memory=5Mi
+```
+
+Initially we will only support CPU and memory, but will eventually support more resources. See
+[#16889](https://github.com/kubernetes/kubernetes/pull/16889) for disk accounting.
+
+If SystemReserved is not set it defaults to 0. Under this condition `Allocatable == Capacity`, and
+the system behavior is equivalent to the 1.1 behavior with scheduling based on Capacity.
+
+#### Reserved
+
+In the initial implementation, `Reserved` will be functionally equivalent to
+[`SystemReserved`](#system-reserved), but with a different semantic meaning. While system reserved
+designates resources set asside for system components, Reserved designates resources set asside for
+non-system components running outside of Kubernetes (currently this is reported as all the processes
+lumped together in the `/system` raw container).
+
+#### Allocatable flag
+
+If the `--allocatable` flag is set on the kubelet, then its value overrides the
+[`Allocatable`](#allocatable) calculation and is always reported instead. Initially this value can
+only be set at Kubelet start-up time, but will [eventually](#future-work) be converted to an
+updatable Config API (per-kubelet) parameter. Its value is specified using the same format as the
+[system reserved](#system-reserved) flag.
```

Comment:
I occurs to me, having written the above, that that might be mischaracterizing our needs.

@sttts Do we need to grow the resources allocated to the Kubelet at runtime to accommodate additional scheduled Mesos task resources so that Kubernetes can enforce them? Or do we want to just give the Kubelet all of the resources allocable by the Mesos agent and let Mesos handle telling the Executor to kill tasks?